src/layer/x86/shufflechannel_x86.cpp

```diff
@@ -387,6 +402,21 @@ int ShuffleChannel_x86::forward(const Mat& bottom_blob, Mat& top_blob, const Opt
                     ptr1 += 8;
                     outptr += 8;
                 }
+
+                for (int i = 0; i < 8; i++)
+                {
+                    if (i % 2)
+                    {
+                        *outptr = *ptr1;
+                        ptr1 += 1;
+                    }
+                    else
+                    {
+                        *outptr = *ptr0;
+                        ptr0 += 1;
+                    }
+                    outptr += 1;
+                }
```

Comment:
I don't think there is a risk of out of range reading in some of the pack8/pack16 codes.